### ISSUE TITLE
No typescript declaration files for locales

### ISSUE THREAD

**Issue by IanVS** [g45t345rt/minifaker#6]
In the published module, there are no typescript definitions for the locales.  This is a problem because I am unable to use the side-effect import of `'minifaker/dist/locales/en'` to set a locale, for some reason.  It only seems to work if I follow this pattern:

```
import { addLocale, word, WordType } from 'minifaker';
import en from 'minifaker/dist/locales/en';

addLocale('en', en);
```

But then I get a typescript error.

**Comment by g45t345rt** [comment]
Typescript for locales are not generated because you don't have to use `addLocale`.

```ts
import { word, WordType } from 'minifaker'
import 'minifaker/dist/locales/en'

word()
```

Here is where it's added when importing the locale
https://github.com/g45t345rt/minifaker/blob/242926ef0e761bc458a78c115b1e5722496ccb48/src/locales/en/index.ts#L71

The `addLocale` function is if you want to create your own custom locale.

What is the Typescript error?

**Comment by IanVS** [comment]
I don't get a typescript error if I use the approach you mention, with the side-effect import.  I get an error from `word` that a locale has not been set.

I'll try to make a reproduction to share.

**Comment by g45t345rt** [comment]
Added type for locale

![a6c1745be94fb3ffe9e4ee8bd4944637](https://user-images.githubusercontent.com/49697006/148655023-3d57256e-1bd5-412f-b616-ffd399ccbc73.png)

**Comment by IanVS** [comment]
Thanks!  Here's a reproduction of what I see when using the side-effect import: 

https://stackblitz.com/edit/node-z2ajob?file=index.ts

**Comment by IanVS** [comment]
I tried out the new version, and I still get an error, I think because the module is not defined as having a `Locale` for a default export.

![image](https://user-images.githubusercontent.com/4616705/148783364-a6377bc0-01b7-4636-a741-79c7d5c62adf.png)

**Comment by g45t345rt** [comment]
Thanks a lot for the repro! I was able to fix it.

I had to seperate both syntax in there own unique folder `dist/esm`, `dist/cjs`. The exact problem was that the `addLocale` while  using side-effect import was resolving to the cjs file and the `word` function was using the esm file. They don't share the same locales :S

Plus, I added the NPM exports for locales. You don't have to use `dist` anymore :)
```ts
import 'minifaker/locales/en'
```

https://github.com/g45t345rt/minifaker/blob/6664ed5e0563bd533cbc7b32e08cf394ef90dbff/package.json#L8-L28
